### PR TITLE
Hotfix/2017 01 04 iss1089 edit by query

### PR DESCRIPTION
**Pull request by Steven-Eardley**
#1089 - there are lots of URLs in these objects, this script runs over the JSON dumped objects, uses a regex to change ```http://www.hindawi``` to ```https://www.hindawi``` wherever it occurs, then bulk saves them. It can be run on journals or suggestions if supplied with the correct query.

It's pretty low-level, so there's some danger if this is used incorrectly. A safer but slower approach would be editing all the URLs via the formcontext.

Also, it could be handy to write the counterpart to this; roll back records to a previous version based on a query.

### CODE REVIEW

File: portality/migrate/1089_edit_field_by_query/regex_http_to_https.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        exit(1)
+
+    if not args.type:
+        print 'One or more type parameters are required.'
+        exit(1)
+
+    print 'Starting {0}.'.format(datetime.now())
+
+    # Connection to the ES index
+    conn = raw.make_connection(None, 'localhost', 9200, 'doaj')
```
Comment (emanuil-tolev) [inline portality/migrate/1089_edit_field_by_query/regex_http_to_https.py]: This won't work on production I'm afraid, it overrides the ELASTICSEARCH_HOST variable to use a different hostname, "gateway". It's best if the scripts respect the configuration variables from settings anyway.

Esprit should have support for the all-in-one Elasticsearch connection url (i.e. with schema, host and port all in one string) that DOAJ defines. This, or something close to it, should work:

```python
conn = esprit.raw.Connection(app.config.get("ELASTIC_SEARCH_HOST"), app.config.get("ELASTIC_SEARCH_DB"))
```

Comment (emanuil-tolev) [inline portality/migrate/1089_edit_field_by_query/regex_http_to_https.py]: That is in fact the line used by a recent migration, `portality/migrate/972_appl_form_changes/appl_form_changes.py:13`, so it should work as-is.

**Comment by emanuil-tolev** [comment]
When merging, it will be best to merge the hotfix branch into develop first, without merging it into production. Then we can test at our leisure on testdoaj.cottagelabs.com, and only fully merge it into production with git flow when we are about to deploy it.

**Comment by richard-jones** [comment]
Yup, as long as we can test this in the test environment first, all looks fine to me

**Pull request merged**